object_storage_api/routers/image.py

```diff
@@ -67,3 +68,20 @@ def get_images(
         logger.debug("Primary filter: '%s'", primary)
 
     return image_service.list(entity_id, primary)
+
+
+@router.get(path="/{image_id}", summary="Get an image by ID", response_description="Single image")
+def get_image(
+    image_id: Annotated[str, Path(description="The ID of the item to get")], image_service: ImageServiceDep
+) -> ImageGetUrlInfoSchema:
+    # pylint: disable=missing-function-docstring
+    logger.info("Getting image with ID %s", image_id)
+    message = "An image with such ID was not found"
+    try:
+        image = image_service.get(image_id)
+        if not image:
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message)
+        return ImageGetUrlInfoSchema(**image.model_dump())
+    except InvalidObjectIdError as exc:
+        logger.exception("The ID is not a valid ObjectId value")
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=message) from exc
```

Comment:
This doesn't need to be repeated due to the exception middleware already detecting any exceptions inheriting from `BaseAPIException`. Same for the one above depending on what we decide to do about the error in the service layer.